nuxeo-features/nuxeo-platform-mail/nuxeo-platform-mail-core/src/main/java/org/nuxeo/ecm/platform/mail/service/MailServiceImpl.java

```diff
@@ -209,6 +228,14 @@ public MessageActionPipe getPipe(String name) {
     }
 
     protected Session newSession(Properties props) {
-        return EmailHelper.newSession(props);
+        // build a key for sessions cache
+        String sessionKey = props.entrySet()
+                                 .stream()
+                                 // by design of PropertiesFetcher, keys are strings
+                                 // sort them alphabetically to have a deterministic order
+                                 .sorted(Comparator.comparing(e -> e.getKey().toString()))
+                                 .map(e -> e.getKey() + "#" + e.getValue())
```

Comment:
Wouldn't it be simpler to sort after the map, when you know you have a String already?